axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceTermServiceImpl.java

```diff
@@ -871,4 +874,20 @@ public void unselect(InvoiceTerm invoiceTerm) throws AxelorException {
       invoiceTermRepo.save(invoiceTerm);
     }
   }
+
+  @Override
+  @Transactional
+  public void computeTotalPaymentSession(PaymentSession paymentSession) {
+    BigDecimal sessionTotalAmount =
+        invoiceTermRepo
+            .all()
+            .filter(
+                "self.paymentSession = :paymentSession AND self.isSelectedOnPaymentSession = TRUE")
+            .bind("paymentSession", paymentSession.getId())
+            .fetchStream()
+            .map(InvoiceTerm::getAmountPaid)
+            .reduce(BigDecimal.ZERO, BigDecimal::add);
+    paymentSession.setSessionTotalAmount(sessionTotalAmount);
+    Beans.get(PaymentSessionRepository.class).save(paymentSession);
+  }
```

Comment:
* This method should be in paymentSessionService, please move it (create PaymentSessionService if needed).
* Also please change the method to directly sum in the query (example: https://github.com/axelor/axelor-open-suite/blob/924c3a995c67a527b4f054f7136b1961dd8721e9/axelor-account/src/main/java/com/axelor/apps/account/service/AccountingReportServiceImpl.java#L481)